### ISSUE TITLE
Invalid metadata error when installing with pip>24.0

### ISSUE THREAD

**Issue by Hillard28** [geopolars/geopolars#243]
When attempting to install geopolars via `pip>24.0`, geopolars metadata is considered invalid given it's usage of the wildcard operator `.*` and `>=`. For `pip>24.0`, `.*` can only be used with `==` or `!=`.

To reproduce (on Fedora Linux),
```
python -m venv ~/gpltest
source ~/gpltest/bin/activate
python -m pip install pip --upgrade
python -m pip install geopolars
```

The exact error is (here for 0.1.0a4, but the installer will cycle through the different versions and throw the same error)
```
WARNING: Ignoring version 0.1.0a4 of geopolars since it has invalid metadata:
Requested geopolars from https://files.pythonhosted.org/packages/4b/4a/cc4c96b9c2604f99bfcc618273db623624239f1519e05e604b211d11e269/geopolars-0.1.0a4-cp38-abi3-manylinux_2_24_x86_64.whl has invalid metadata: .* suffix can only be used with `==` or `!=` operators
    pyarrow>=4.0.*
           ~~~~~~^
Please use pip<24.1 if you need to use this version.
```

Using `pip<24.1` solves the issue, but it would be nice to have support for recent versions.

**Comment by kylebarron** [comment]
GeoPolars is not currently functional, regardless. We need Polars to enable additional APIs to make GeoPolars possible. See https://github.com/pola-rs/polars/issues/1830#issuecomment-2218102856

**Comment by Hillard28** [comment]
Ah, I see. For some reason I thought I'd recalled using this package in the past.

**Comment by kylebarron** [comment]
Closing as stale; now that Polars extension types are implemented, this project is now unblocked (https://github.com/pola-rs/geopolars/issues/245)
